app/controllers/topics_controller.rb

```diff
@@ -6,15 +6,30 @@ def index
 
   def show
     @topic = policy_scope(Topic).find(params[:id])
-    @signs = policy_scope(@topic.signs)
+
+    search_results ||= TopicSignService.new(search:, relation: policy_scope(Sign), topic: @topic).process
+    @signs = search_results.data
+    @page = search_results.support
+
     authorize @topic
   end
 
   def uncategorised
+    # "uncategorised" isn't a topic
+    # it references signs that LACK a topic
     @topic = Topic.new(name: Topic::NO_TOPIC_DESCRIPTION)
+    @search_results ||= TopicSignService.new(search:, relation: policy_scope(Sign), topic: @topic).process
+    @signs = @search_results.data
+    @page = @search_results.support
```

Comment:
suggestion: this can be a local variable

```suggestion
    search_results = TopicSignService.new(search:, relation: policy_scope(Sign), topic: @topic).process
    @signs = search_results.data
    @page = search_results.support
```